src/components/pages/Reserve.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Swal from 'sweetalert2';
+import { selectUser } from '../../redux/usersession/usersessionsSlice';
+import { createReservation } from '../../redux/reservations/reservationsSlice';
+
+function Reserve() {
+  const [boats, setBoats] = useState([]);
+  const user = useSelector(selectUser);
+  const [username, setUsername] = useState(user.name);
+  const [city, setCity] = useState('');
+  const [date, setDate] = useState('');
+  const [boat_id, setBoatId] = useState(undefined);
+  const dispatch = useDispatch();
+  console.log(boats.data);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    await dispatch(
+      createReservation({
+        username,
+        city,
+        date,
+        boat_id,
```

Comment:
Here I would suggest using the required boat_id: boatId to create the Reservation, please update it.